### PR TITLE
feat: [CDS-40961]: Improve error handling for azure web app

### PR DESCRIPTION
**Pull request by abosii-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35954)

### CODE REVIEW

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/azure/exception/AzureAppServicesRuntimeExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  private static final String HINT_WEBAPP_NOT_FOUND =
+      "Check to make sure web app '%s' exists under resource group '%s'";
+  private static final String HINT_DEPLOYMENT_SLOT_NOT_FOUND =
+      "Check to make sure you have provided correct deployment slot '%s' for web app '%s'";
+  private static final String HINT_INVALID_ARTIFACT_TYPE = "Check if deployed artifact '%s' is packaged '%s' file";
+  private static final String HINT_CONFIGURE_WEBSISTES_PORT =
+      "If container listens to a different port than 80 or 8080 configure WEBSISTES_PORT in application settings";
+  private static final String HINT_CHECK_CONTAINER_LOGS = "Check container logs for more details";
+
+  private static final String EXPLANATION_WEBAPP_NOT_FOUND =
```
Comment (vlica-harness) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/azure/exception/AzureAppServicesRuntimeExceptionHandler.java:49]: nit: for WEBAPP_NOT_FOUND saw you didn't mention subscription, but for DEPLOYMENT_SLOT_NOT_FOUND is mentioned

**Review by vlica-harness — APPROVED** [review]
> Reviewed 21 of 21 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35954)* status: :shipit: complete! ✓

**Review by tmacari-harness — DISMISSED** [review]
> Reviewed 21 of 21 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35954)* status: :shipit: complete! ✓

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 6 Issues 
* Vulnerability 1 Vulnerabilities 
 * Code issue 5 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/35954)

**Comment by abosii-harness** [comment]
trigger messagecheck

**Review by vlica-harness — APPROVED** [review]
> Reviewed 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35954)* status: :shipit: complete! ✓

**Review by anilchowdhury-harness — DISMISSED** [review]
:lgtm:

> Reviewed 17 of 21 files at r1, 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35954#-:-N8hKL1s2nmzb9f_zHVT:bnfp4nl)* status: :shipit: complete! ✓

**Review by vlica-harness — DISMISSED** [review]

**Review by tmacari-harness — CHANGES_REQUESTED** [review]
> Reviewed 1 of 4 files at r2, 2 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35954)* status: 1 file(s) to review. 1 change requests. 

___
*[`260-delegate/src/test/resources/kryo-registrations.txt` line 2257 at r3](https://reviewable.io/reviews/harness/harness-core/35954#-N8h_BqrCA6uOA-SkSgS:-N8h_BqrCA6uOA-SkSgT:bpm97k0) ([raw file](https://github.com/harness/harness-core/blob/56edcb8020b5096b1f181f85c79efcc381064248/260-delegate/src/test/resources/kryo-registrations.txt#L2257)):*
> ```help
> 980030:io.harness.exception.InterruptedRuntimeException
> 980031:io.harness.exception.InvalidIdentifierRefException
> 980033:io.harness.exception.ngexception.AzureAppServiceTaskException
> ```

it is registered with 980032

**Review by tmacari-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35954)* status: :shipit: complete! ✓

**Review by anilchowdhury-harness — APPROVED** [review]
:lgtm:

> Reviewed 2 of 3 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35954#-:-N8hmkIj0LUlB4zk3s8o:bnfp4nl)* status: :shipit: complete! ✓

**Comment by abosii-harness** [comment]
trigger codebasehash

**Pull request merged**